kura/org.eclipse.kura.ai.triton.server/src/main/java/org/eclipse/kura/ai/triton/server/TritonServerServiceImpl.java

```diff
@@ -0,0 +1,766 @@
+import static java.util.Objects.nonNull;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.KuraIOException;
+import org.eclipse.kura.ai.inference.InferenceEngineService;
+import org.eclipse.kura.ai.inference.ModelInfo;
+import org.eclipse.kura.ai.inference.ModelInfoBuilder;
+import org.eclipse.kura.ai.inference.Tensor;
+import org.eclipse.kura.ai.inference.TensorDescriptor;
+import org.eclipse.kura.ai.inference.TensorDescriptorBuilder;
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.eclipse.kura.executor.CommandExecutorService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.ProtocolStringList;
+
+import inference.GRPCInferenceServiceGrpc;
+import inference.GRPCInferenceServiceGrpc.GRPCInferenceServiceBlockingStub;
+import inference.GrpcService.InferParameter;
+import inference.GrpcService.InferTensorContents;
+import inference.GrpcService.ModelInferRequest;
+import inference.GrpcService.ModelInferResponse;
+import inference.GrpcService.ModelInferResponse.InferOutputTensor;
+import inference.GrpcService.ModelMetadataRequest;
+import inference.GrpcService.ModelMetadataResponse;
+import inference.GrpcService.ModelReadyRequest;
+import inference.GrpcService.ModelReadyResponse;
+import inference.GrpcService.RepositoryIndexRequest;
+import inference.GrpcService.RepositoryIndexResponse;
+import inference.GrpcService.RepositoryIndexResponse.ModelIndex;
+import inference.GrpcService.RepositoryModelLoadRequest;
+import inference.GrpcService.RepositoryModelUnloadRequest;
+import inference.GrpcService.ServerLiveRequest;
+import inference.GrpcService.ServerLiveResponse;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+
+public class TritonServerServiceImpl implements InferenceEngineService, ConfigurableComponent {
+
+    private static final Logger logger = LoggerFactory.getLogger(TritonServerServiceImpl.class);
+
+    private CommandExecutorService commandExecutorService;
+    private TritonServerServiceOptions options;
+    private TritonServerLocalManager tritonServerLocalManager;
+
+    private ManagedChannel grpcChannel;
+    private GRPCInferenceServiceBlockingStub grpcStub;
+
+    public void setCommandExecutorService(CommandExecutorService executorService) {
+        if (isNull(this.commandExecutorService)) {
+            this.commandExecutorService = executorService;
+        }
+    }
+
+    public void unsetCommandExecutorService(CommandExecutorService executorService) {
```

Comment:
If this reference is static, specifying the unbind method is probably not necessary